src/wp-includes/post.php

```diff
@@ -3089,6 +3089,17 @@ function wp_count_posts( $type = 'post', $perm = '' ) {
 function wp_count_attachments( $mime_type = '' ) {
 	global $wpdb;
 
+	$cache_key = sprintf(
+		'attachments%s',
+		! empty( $mime_type ) ? ':' . str_replace( '/', '_', implode( '-', (array) $mime_type ) ) : ''
+	);
+
+	$counts = wp_cache_get( $cache_key, 'counts' );
+	if ( false !== $counts ) {
+		/** This filter is documented in wp-includes/post.php */
+		return apply_filters( 'wp_count_attachments', $counts, $mime_type );
+	}
```

Comment:
It _might_ better to remove the early return and instead wrap the caching with `if ( false === $counts ) {}`.

This would mean only one `apply_filters( 'wp_count_attachments' )` needs to maintain in this function and creates less room for mistakes later.

That said, it means indenting the caching lines.

What are your thoughts on this? Better to return early and have two `apply_filters( 'wp_count_attachments' )`, or better to wrap the caching in `if ( false === $counts ) {}`?